app/main.py

```diff
@@ -1,2 +1,98 @@
+from __future__ import annotations
+from typing import Hashable, Any, Generator
+
+
+class KeyValuePair:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self._key = key
+        self.value = value
+
+    def __eq__(self, other: KeyValuePair | int):
+        if isinstance(other, KeyValuePair):
+            return other.key == self.key
+        return other == self.key
+
+    def __hash__(self):
+        return hash(self.key)
+
+    def __str__(self) -> str:
+        return f"Key={self.key} : Value={self.value}"
+
+    @property
+    def key(self):
+        return self._key
+
+
+# I implement dictionary via chaining
 class Dictionary:
-    pass
+    _INITIAL_CAPACITY = 8
+    _LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self._hash_table = [[] for _ in range(Dictionary._INITIAL_CAPACITY)]
```

Comment:
Also, I think that here `[[]] * self._INITIAL_CAPACITY` would be a better way to do it. Ignore this comment if you disagree.

Reply:
Thanks for comment, but it doesn't work in such way because all list will have the same reference.

Reply:
Whoops. I forgot about that.